stoobly_agent/app/cli/helpers/endpoints_import_service.py

```diff
@@ -124,14 +145,19 @@ def import_response_header_name(project_id: int, endpoint_id: int, response_head
     ENDPOINT_COMPONENT_NAMES[4]: import_response_param_name
 }
 
-def process_import(component_name: str, *args):
-    return component_name_import_dispatch[component_name](*args)
+def process_import(component_name: str, body_param_parents: Dict[int, int], response_param_parents: Dict[int, int], *args):
+    if component_name == ENDPOINT_COMPONENT_NAMES[1]:
+        return component_name_import_dispatch[component_name](*args, body_param_parents)
+    elif component_name == ENDPOINT_COMPONENT_NAMES[4]:
+        return component_name_import_dispatch[component_name](*args, response_param_parents)
+    else:
+        return component_name_import_dispatch[component_name](*args)
 
-def import_component_names(project_id: int, endpoint_id: int, endpoint: EndpointShowResponse, resources: Dict[str, EndpointsResource]):
+def import_component_names(project_id: int, endpoint_id: int, endpoint: EndpointShowResponse, resources: Dict[str, EndpointsResource], body_param_parents: Dict[int, int], response_param_parents: Dict[int, int]):
     for component_name in ENDPOINT_COMPONENT_NAMES:
         for component in endpoint.get(f'{component_name}s', {}):
             resource = resources[component_name]
-            process_import(component_name, project_id, endpoint_id, resource, component)
+            process_import(component_name, body_param_parents, response_param_parents, project_id, endpoint_id, resource, component)
```

Comment:
If the response or body is more than one level deep, does this handle the recursion?

Reply:
No, `endpoint['response_param_names']` and `endpoint['body_param_names']` are lists so theres no recursion here. The `body_param_names` list for `POST /v2/pets` in `petstore-references.yaml` would look like this (This one has a request body that is more than one level deep):

```
[
    {'endpoint_id': 2, 'name': 'base', 'query': 'base', 'is_required': False, 'inferred_type': 'Hash', 'is_deterministic': True, 'id': 1, 'body_param_name_id': None}
    {'endpoint_id': 2, 'name': 'name', 'query': 'base.name', 'is_required': True, 'inferred_type': 'String', 'is_deterministic': True, 'id': 2, 'body_param_name_id': 1}
    {'endpoint_id': 2, 'name': 'tag', 'query': 'base.tag', 'is_required': False, 'inferred_type': 'String', 'is_deterministic': True, 'id': 3, 'body_param_name_id': 1}
    {'endpoint_id': 2, 'name': 'extra', 'query': 'extra', 'is_required': False, 'inferred_type': 'Hash', 'is_deterministic': True, 'id': 4, 'body_param_name_id': None}
    {'endpoint_id': 2, 'name': 'color', 'query': 'extra.color', 'is_required': True, 'inferred_type': 'String', 'is_deterministic': True, 'id': 5, 'body_param_name_id': 4}
    {'endpoint_id': 2, 'name': 'age', 'query': 'extra.age', 'is_required': True, 'inferred_type': 'Integer', 'is_deterministic': True, 'id': 6, 'body_param_name_id': 4}
    {'endpoint_id': 2, 'name': 'adoption', 'query': 'extra.adoption', 'is_required': False, 'inferred_type': 'Hash', 'is_deterministic': True, 'id': 7, 'body_param_name_id': 4}
    {'endpoint_id': 2, 'name': 'adopted', 'query': 'extra.adoption.adopted', 'is_required': True, 'inferred_type': 'Boolean', 'is_deterministic': True, 'id': 8, 'body_param_name_id': 7}
    {'endpoint_id': 2, 'name': 'shelter', 'query': 'extra.adoption.shelter', 'is_required': False, 'inferred_type': 'String', 'is_deterministic': True, 'id': 9, 'body_param_name_id': 7}
    {'endpoint_id': 2, 'name': 'id', 'query': 'id', 'is_required': True, 'inferred_type': 'Integer', 'is_deterministic': True, 'id': 10, 'body_param_name_id': None}
]
```

<img width="1151" alt="Screen Shot 2024-05-19 at 10 35 55 AM" src="https://github.com/Stoobly/stoobly-agent/assets/58492699/c7597f59-af50-4694-9c36-c42b9808ef8d">

The `id` and `parent_id` values are used as keys in the `body_param_parents` and `response_param_parents` dicts and each map to the `id` value returned in the HTTP responses. This way we can look up the right `response_param_name_id` or `body_param_name_id` for each response/body param name resource by using the `parent_id`.

The recursive stuff is handled in the OAS adapter.